### DISCUSSION TITLE
Is there a plan to implement multivariate forecasting in Time Series Forecasting?

### DISCUSSION THREAD

**Discussion by mreismendes** [autogluon/autogluon#2489] [Q&A]
Is there a plan to implement multivariate forecasting in Time Series Forecasting? Models like: "gpvar" and "deepvar".

Thank you

**Comment by tonyhoo** [comment]
Hi @mreismendes , thanks for your interests and feedback. While multivariate forecasting models are not part of the AutoGluon TS model zoo now. There is a workaround if you want to pass dependent time series (dynamic time varying features such as holidays and temperatures) for forecasting purpose. You can checkout this [tutorial](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#known-covariates) to see if that meet your forecasting needs. Thanks

**Reply by mreismendes** [reply]
Hi @tonyhoo, thank you for the reply. In this specific case, I know there are interactions between those variables. Is there a plan in a foreseeable future to get those multivariate models in AutoGluon TS model zoo?

**Reply by tonyhoo** [reply]
@canerturkmen will be able to help to address any roadmap question regarding TS modules. Meanwhile, if there are interactions between variables, you can try to pass [known covariates](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#known-covariates) or [static features](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#static-features) so that they can be encoded and pass to underlying models such as [DeepAR](https://ts.gluon.ai/stable/api/gluonts/gluonts.model.deepar.html)

**Comment by langslike** [comment]
@tonyhoo but the TimeSeriesPredictor.predict requires that known_covariates is the future data known in advance, which is not available in many cases

in the example below, what if i only have the known_covariates on the same date in 'data'， but don`t have the future_known_covariates ,  can I still do the predict? do I need to adjust the data? Do I need a shift on known_covariates training data to account for that?

Examples from autogluon/timeseries/predictor.py
        --------
        >>> print(data)
                            target  promotion  price
        item_id timestamp
        A       2020-01-05      20          0   19.9
                2020-01-06      40          1    9.9
                2020-01-07      32          0   15.0
        B       2020-03-01      13          0    5.0
                2020-03-02      44          1    2.9
                2020-03-03      72          1    2.9
        >>> predictor = TimeSeriesPredictor(prediction_length=2, known_covariates_names=["promotion", "price"]).fit(data)
        >>> print(future_known_covariates)
                            promotion  price
        item_id timestamp
        A       2020-01-08          1   12.9
                2020-01-09          1   12.9
        B       2020-03-04          0    5.0
                2020-03-05          0    7.0
        >>> predictor.predict(data, known_covariates=future_known_covariates)
                            target
        item_id timestamp
        A       2020-01-08      30
                2020-01-09      27
        B       2020-03-04      17
                2020-03-05       8
        """
